presto-main/src/test/java/io/prestosql/server/security/oauth2/TestingHydraService.java

```diff
@@ -0,0 +1,183 @@
+
+    private final FixedHostPortGenericContainer<?> hydraContainer = createHydraContainer()
+            .withNetworkAliases("hydra")
+            .withExposedPorts(4444, 4445)
+            .withEnv("SECRETS_SYSTEM", generateSecret())
+            .withEnv("DSN", DSN)
+            .withEnv("URLS_SELF_ISSUER", "http://hydra:4444/")
+            .withEnv("URLS_CONSENT", "http://consent:3000/consent")
+            .withEnv("URLS_LOGIN", "http://consent:3000/login")
+            .withEnv("OAUTH2_ACCESS_TOKEN_STRATEGY", "jwt")
+            .withEnv("TTL_ACCESS_TOKEN", TTL_ACCESS_TOKEN_IN_SECONDS + "s")
+            .withCommand("serve all --dangerous-force-http")
+            .waitingFor(Wait.forHttp("/health/ready").forPort(4444).forStatusCode(200));
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+
+    TestingHydraService()
+    {
+        closer.register(network);
+        closer.register(databaseContainer);
+        closer.register(migrationContainer);
+        closer.register(consentContainer);
+        closer.register(hydraContainer);
+    }
+
+    public void start()
+    {
+        databaseContainer.start();
+        migrationContainer.start();
+        consentContainer.start();
+        hydraContainer.start();
+    }
+
+    public FixedHostPortGenericContainer<?> createHydraContainer()
+    {
+        return new FixedHostPortGenericContainer<>(HYDRA_IMAGE).withNetwork(network);
+    }
+
+    public Network getNetwork()
+    {
+        return network;
+    }
+
+    public int getHydraPort()
+    {
+        return hydraContainer.getMappedPort(4444);
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        try {
+            closer.close();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String generateSecret()
+    {
+        byte[] randomBytes = new byte[32];
+        SECURE_RANDOM.nextBytes(randomBytes);
+        return Base64.getEncoder().encodeToString(randomBytes);
+    }
+
+    public static void main(String[] args)
+            throws Exception
+    {
+        try (TestingHydraService service = new TestingHydraService()) {
+            // expose containers ports & override environment variables
+            service.consentContainer
+                    .withFixedExposedPort(9020, 3000);
+            service.hydraContainer
+                    .withFixedExposedPort(9001, 4444)
+                    .withFixedExposedPort(9002, 4445)
+                    .withEnv("URLS_SELF_ISSUER", "http://localhost:9001/")
+                    .withEnv("URLS_CONSENT", "http://localhost:9020/consent")
+                    .withEnv("URLS_LOGIN", "http://localhost:9020/login")
+                    .withEnv("TTL_ACCESS_TOKEN", "30m");
+            service.start();
+
+            // create a client for Presto in Hydra
+            service.createHydraContainer()
```

Comment:
This code is duplicated with `TestOAuth2WebUiAuthenticationFilter`. Would it make sense to add a method on `TestingHydraService` to do this?